integrations/opsgenie/main.py

```diff
@@ -33,31 +39,42 @@ async def enrich_services_with_team_data(
         return service
 
 
-async def enrich_incident_with_alert_data(
+async def enrich_incidents_with_impacted_services(
     opsgenie_client: OpsGenieClient,
     semaphore: asyncio.Semaphore,
-    incident: dict[str, Any],
-) -> dict[str, Any]:
-    async with semaphore:
-        if not incident["impactedServices"]:
-            return incident
-        impacted_services = await opsgenie_client.get_impacted_services(
-            incident["impactedServices"]
+    incidents: list[dict[str, Any]],
+    chunk_size: int = INCIDENT_CHUNK_SIZE,
+) -> list[dict[str, Any]]:
+    enriched_incidents = []
+
+    for incident_chunk in chunked(incidents, chunk_size):
+        impacted_service_ids = set()
+        incident_to_service_map = defaultdict(list)
+
+        for incident in incident_chunk:
+            if incident["impactedServices"]:
+                impacted_service_ids.update(incident["impactedServices"])
+                incident_to_service_map[incident["id"]] = incident["impactedServices"]
+        logger.info(
+            f"Got {len(impacted_service_ids)} unique impacted services from {len(incident_chunk)} incidents with a chunk size of {chunk_size}"
         )
-        incident["__impactedServices"] = impacted_services
-        return incident
 
+        # Fetch all impacted services for this chunk in one API call
+        async with semaphore:
+            all_impacted_services = await opsgenie_client.get_impacted_services(
+                list(impacted_service_ids)
+            )
```

Comment:
if we do the relation based on `id` we don't need to fetch all of these entities anymore